core/variables.js

```diff
@@ -270,6 +270,55 @@ Blockly.Variables.createVariable = function(workspace, opt_callback) {
 };
 
 /**
+ * Rename a variable with the given workspace, variableType, and oldName.
+ * @param {!Blockly.Workspace} workspace The workspace on which to rename the
+ *     variable.
+ * @param {function(?string=)=} opt_callback A callback. It will
+ *     be passed an acceptable new variable name, or null if change is to be
+ *     aborted (cancel button), or undefined if an existing variable was chosen.
+ * @param {string} variableType Optional variable type, like 'string' or 'list'.
+ * @param {string} currentName Name of the variable being renamed.
+ */
+Blockly.Variables.renameVariable = function(workspace, opt_callback,
+  variableType, currentName) {
+  var promptAndCheckWithAlert = function(defaultName) {
```

Comment:
Could you add a comment above this declaration that the function needs to be named so it can be called recursively? I had to ask Andrew (who wrote it) to explain why and it took him a few minutes to remember. =)